### PR TITLE
Refactor defaults/merger

### PR DESCRIPTION
**Pull request by jhugman**
Relates to [ EXP-4110](https://mozilla-hub.atlassian.net/browse/EXP-4110).

According to [The Big O of Code Reviews](https://www.egorand.dev/the-big-o-of-code-reviews/), this is a O(_n_) change.

This small refactor extracts methods the defaults merger into smaller methods. It doesn't rewrite anything (so might be a O(_1_)) but just splits out the `overwrite_defaults` code into its own method.

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

File: components/support/nimbus-fml/src/defaults/merger.rs [hunk 1]
```diff
@@ -184,41 +222,29 @@ impl<'object> DefaultsMerger<'object> {
...
+            Ok(merged)
+        } else {
+            Ok(empty_object)
         }
-        Ok(())
+    }
+
+    /// A convenience method to get the defaults from the feature, and merger it
+    /// with the passed value.
+    pub(crate) fn _merge_feature_config(
```
Comment (jhugman) [inline components/support/nimbus-fml/src/defaults/merger.rs:241]: This method is used later on in EXP-4110, but grouping this together here.

File: components/support/nimbus-fml/src/defaults/merger.rs [hunk 2]
```diff
@@ -174,6 +174,44 @@ impl<'object> DefaultsMerger<'object> {
         feature_def: &mut FeatureDef,
         defaults: &Option<Vec<DefaultBlock>>,
     ) -> Result<(), FMLError> {
+        let variable_defaults = self.collect_feature_defaults(feature_def)?;
```
Comment (jeddai) [inline components/support/nimbus-fml/src/defaults/merger.rs:177]: Are there pre-existing tests covering these changes?

**Review by jeddai — APPROVED** [review]
r+wc

File: components/support/nimbus-fml/src/defaults/merger.rs (hunk 2 above)
Comment (jhugman) [reply components/support/nimbus-fml/src/defaults/merger.rs:177]: Oh yes! The tests covering this code start [here](https://github.com/mozilla/application-services/blob/2b05e22f2e2d5f2f8f4eb4573244061b91700240/components/support/nimbus-fml/src/defaults/merger.rs#L345).

**Pull request merged**